make/langtools/tools/flagsgenerator/FlagsGenerator.java

```diff
@@ -0,0 +1,160 @@
+import com.sun.source.util.JavacTask;
+import com.sun.source.util.TreePath;
+import com.sun.source.util.Trees;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.util.ElementFilter;
+import javax.tools.ToolProvider;
+
+public class FlagsGenerator {
+    public static void main(String... args) throws IOException {
+        var compiler = ToolProvider.getSystemJavaCompiler();
+
+        try (var fm = compiler.getStandardFileManager(null, null, null)) {
+            JavacTask task = (JavacTask) compiler.getTask(null, null, d -> {}, null, null, fm.getJavaFileObjects(args[0]));
+            Trees trees = Trees.instance(task);
+            CompilationUnitTree cut = task.parse().iterator().next();
+
+            task.analyze();
+
+            TypeElement clazz = (TypeElement) trees.getElement(new TreePath(new TreePath(cut), cut.getTypeDecls().get(0)));
+            Map<Integer, List<String>> flag2Names = new TreeMap<>();
+            Map<FlagTarget, Map<Integer, List<String>>> target2FlagBit2Fields = new HashMap<>();
+            Map<String, String> customToString = new HashMap<>();
+            Set<String> noToString = new HashSet<>();
+
+            for (VariableElement field : ElementFilter.fieldsIn(clazz.getEnclosedElements())) {
+                String flagName = field.getSimpleName().toString();
+                for (AnnotationMirror am : field.getAnnotationMirrors()) {
+                    switch (am.getAnnotationType().toString()) {
+                        case "com.sun.tools.javac.code.Flags.Use" -> {
+                            long flagValue = ((Number) field.getConstantValue()).longValue();
+                            int flagBit = 63 - Long.numberOfLeadingZeros(flagValue);
+
+                            flag2Names.computeIfAbsent(flagBit, _ -> new ArrayList<>())
+                                      .add(flagName);
+
+                            List<?> originalTargets = (List<?>) valueOfValueAttribute(am);
+                            originalTargets.stream()
+                                           .map(value -> FlagTarget.valueOf(value.toString()))
+                                           .forEach(target -> target2FlagBit2Fields.computeIfAbsent(target, _ -> new HashMap<>())
+                                                                                   .computeIfAbsent(flagBit, _ -> new ArrayList<>())
+                                                                                   .add(flagName));
```

Comment:
Instead of a dedicated loop to verify no overlaps, we can make the nested map `Map<Integer, String>` and fail fast whenever we detect a conflict, like:
```java
var oldFlagName = target2FlagBit2Fields.computeIfAbsent(target, _ -> new HashMap<>()).put(flagBit, flagName);
if (oldFlagName != null) {
    // Fail fast code
}
```
I personally don't see a reason to collect all conflicting fields for a flag if any of these conflicts already causes a failure.